### ISSUE TITLE
 DT 3.2.1 - Crop and Rotate unsuable on CR2 in portrait mode

### ISSUE THREAD

**Issue by roumano** [darktable-org/darktable#6191]
**Describe the bug**

The crop and rotate module is unsuable for some CR2 (all of them are in portrait mode)
i can't see in the border of the image, the usual rechteck arround the picture to reduce isn't available.
Sometime it appear outside of the image, without the possiblity to reduce or move
Sometiem but rarely, if i try to reset the crop and module and apply the change on these image, darktable is crash but it's rarely so i wasn't able to get a trace of the crash

**To Reproduce**

**Expected behavior**

Able to reduce the picutre

**Screenshots**
I have blue images but we can see the crop and module totally outside of the picture ...

![Capture d’écran de 2020-09-03 22-10-27](https://user-images.githubusercontent.com/6402894/92165305-20da2280-ee37-11ea-8fb6-efddc0dc3984.png)
![Capture d’écran de 2020-09-03 22-27-59](https://user-images.githubusercontent.com/6402894/92165329-2afc2100-ee37-11ea-9955-908a1f10dacc.png)
![Capture d’écran de 2020-09-03 22-23-01](https://user-images.githubusercontent.com/6402894/92165330-2afc2100-ee37-11ea-9677-a0291d25103f.png)
![Capture d’écran de 2020-09-03 22-28-08](https://user-images.githubusercontent.com/6402894/92165335-2cc5e480-ee37-11ea-9c23-a354d6df56c5.png)
![Capture d’écran de 2020-09-03 22-28-14](https://user-images.githubusercontent.com/6402894/92165339-2fc0d500-ee37-11ea-9a29-7c1ff327d8d8.png)

**Platform (please complete the following information):**
 - Darktable Version: 3.2.1
 - OS: Gentoo Linux
 - OpenCL is "not available" 
 - ATI Radeon HD 7950, driver used : RADEON: Driver for ATI/AMD Radeon chipsets ( vendor="X.Org Foundation")

**Additional context**
 - Can you reproduce with another Darktable version?
No, it's never happen on previous version and now not able to use older version on my computer
 - Can you reproduce with a RAW or Jpeg or both?
Reproduce only with RAW
 - Are the steps above reproduce with a fresh edit (removing history)?
Yes, i have try with 
```
rm -rf /tmp/4 /tmp/db4.test && mkdir /tmp/4 && LANG=C darktable --cachedir /tmp/4 --configdir /tmp/4 --library /tmp/db4.test 
```
 - Attach an XMP if this is necessary
[IMG_2173.CR2.zip](https://github.com/darktable-org/darktable/files/5171109/IMG_2173.CR2.zip)
Even if remove the crop and rotate module in the history ( and compress) , i can reproduce the issue.

 - Is the issue still present using an empty/new config-dir
Yes if use my xmp, without xmp not able to reproduce

**Comment by zaltysz** [comment]
Is this newly imported image or old one? 

Your likely problem is module order. Reset and disable cropping, than move it above orientation module, reload the image and then try to crop. Does it work correctly now?

**Comment by roumano** [comment]
It's a new imported image (tested with a empty darktable config and dB ( `rm -rf /tmp/4 /tmp/db4.test && mkdir /tmp/4 && LANG=C darktable --cachedir /tmp/4 --configdir /tmp/4 --library /tmp/db4.test ` ), but can't reproduce it all the time

**Comment by Hodapp87** [comment]
Running into this too on a portrait image. It's completely unusable.

**Comment by johnny-bit** [comment]
Ah good ol crash and break module ;) I tried reproducting but as it turns out... I don't have any portrait image (or have very few of them). do you have any portrait images that can be used to investigate issue? Also - does the issue still happen on current master?

**Comment by roumano** [comment]
Hi,
Yes i can share a raw files but not with public access like here. where i can send you the raw file ?
Thanks in advance

**Comment by johnny-bit** [comment]
Actually it's better to have a public raw to test, something like play raw on https://discuss.pixls.us/c/processing/playraw/30 (this requires a CC0 licence). That way more than one developer can have same testcase and so on :)

If you don't want to share a raw on playraw then next best thing may be simple test shot of color target in portrait mode or something like this, that you can then confirm that you can replicate the issue on the sample file :)

**Comment by roumano** [comment]
I can't reproduce every time and i'm not sure if it's link to specific condition or not ...

@Hodapp87 , can you share a raw where you are sure it's reproducible ?

**Comment by johnny-bit** [comment]
There've been fixes to crop&rotate module that sanitize it's params so that it's damn near impossible to reproduce now (not that I have been successfull previously)

in dt 3.4.x it's most likely fixed and in 3.6 (coming in less than 6 days) it's certainly fixed. Closing :)

If you somehow manage to reproduce - please comment and make sure to include XMP & RAW file to test with. and I'll re-open.
